src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import measure_theory.measure.measure_space
+
+/-!
+# Weak convergence of (finite) measures
+
+This file will define the topology of weak convergence of finite measures and probability measures
+on topological spaces. The topology of weak convergence is the coarsest topology w.r.t. which
+for every bounded continuous `ℝ≥0`-valued function `f` on, the integration of `f` against the
+measure is continuous.
+
+TODOs:
+* Rename the types `finite_measures α` -> `finite_measure α` and
+  `probability_measures α` -> `probability_measure α` after PR #8831.
+* Define the topologies (the current version only defines the types) via
+  `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`.
+* Prove that an equivalent definition of the topologies is obtained using bounded continuous
+  `ℝ`-valued functions instead.
+* Include the portmanteau theorem on characterizations of weak convergence of (Borel) probability
+  measures.
+
+## Main definitions
+
+The main definitions are the types `finite_measures α` and `probability_measures α`.
+
+TODO:
+* Define the topologies on the above types.
+
+## Main results
+
+None yet.
+
+TODO:
+* Portmanteau theorem.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+The topology of weak convergence of finite Borel measures will be defined using a mapping from
+`finite_measures α` to `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`, inheriting the topology from there.
+
+## References
+
+* [Billingsley, *Convergence of probability measures*][billingsley1999]
+
+## Tags
+
+weak convergence of measures, finite measure, probability measure
+
+-/
+
+noncomputable theory
+open measure_theory
+open set
+open filter
+open_locale topological_space ennreal nnreal
+
+section finite_measures
+/-!
+### Finite measures
+In this section, we define the `Type` of finite measures on a given measurable space. This type
+will be equipped with the topology of weak convergence of measures when the measurable space is
+a topological space equipped with its Borel sigma-algebra.
+-/
+
+variables {α : Type*} [measurable_space α]
+
+/-- Finite measures are defined as the subtype of measures that have the property of being finite
+measures (i.e., their total mass is finite). -/
+def finite_measures (α : Type*) [measurable_space α] : Type :=
```

Comment:
```suggestion
def finite_measures (α : Type*) [measurable_space α] : Type* :=
```
Otherwise lean solves the `*` to mean
```lean
def finite_measures (α : Type) [measurable_space α] : Type :=
```
Which isn't what you want.

Reply:
Thank you! I actually noticed the problem after committing the previous suggestions. I am editing on my computer now, so will not commit suggestions before I push my local changes...

(Usually I use `Type*`, but a few times I've had problems with it, which were solved by changing to `Type`. The choice here was probably an artifact from some such past problem.)

I think together with my local changes I will attempt to merge to master at the same time (there will be renamings such as `finite_measure` -> `is_finite_measure`).

Reply:
Actually, I'll first commit local changes and then in the next step try to merge master. I took into account two of your new suggestions. Thank you!

Reply:
I hope I managed to merge master and do the renamings.

I still have a few review comments I haven't addressed. I will, of course. But some that involve me learning something new might not be done today yet...